src/affordability.ts

```diff
@@ -0,0 +1,169 @@
+import { MonthlyRateDecimal, MonthsPerYear } from './constants';
+import { ComputeMonthlyPaymentInputs } from './payments';
+
+/**
+ * A Cost model for monthly costs
+ */
+export type CostModel = {
+  /**
+   * A factor to scale the present value of the loan by to figure out monthly cost
+   */
+  presentValueFactor: number;
+  /**
+   * A fixed offset of monthly cost to add to the result of the presentValueFactor * presentValue
+   */
+  costOffset: number;
+  /**
+   * A minimum present value for this cost model to be valid
+   */
+  minPresentValue: number;
```

Comment:
To make sure I'm understanding this correctly, in the case of PMI, this value would be the loan amount equivalent to the value defined by the lower bound of the percentage range?

E.g., for loan amount $500_000, PMI rate of 1% for 3-5%, this value would be $15_000?

Reply:
Sort of it's easier to explain with something like taxes:
* Imagine you had a tax rate of 1% for Homes under $1,000,000 and a rate of 1.2% for homes over 1,000,000
* Imagine you had a $200,000 downpayment
* You would create 2 cost model elements, one with a `maxPresentValue` of $800,000 and the other with a `maxPresentValue` of Positive Infinity. The rates would be 0.01 and 0.012 respectively. The cost offset would be $2,000 and $2,400 respectively.